### ISSUE TITLE
Restarting training fails with cpu vs gpu device error

### ISSUE THREAD

**Issue by oliverwm1** [ai2cm/ace#415]
I started a training run using the `ExistingCheckpointConfig` option for the stepper (i.e. starting a new training job from an existing checkpoint). [Beaker link](https://beaker.org/ex/01J4TMK3JNR2WCFBXTX7FNTVAK/tasks/01J4TMK3JWCS7QBGH0PHTA9AR9/job/01J4TMK668MN707H5Y2HWQ0B5A). It failed at the validation step with:

```
37 2024-08-09 04:05:08,803 - root - INFO - Starting Training Loop...
38 2024-08-09 04:05:09,066 - root - INFO - Epoch: 1
39 2024-08-09 04:05:09,067 - root - INFO - Starting training step on epoch 1
40 2024-08-09 04:45:28,770 - root - INFO - Starting validation step on epoch 1
41 Traceback (most recent call last):
42   File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
43     return _run_code(code, main_globals, None,
44   File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
45     exec(code, run_globals)
46   File "/full-model/fme/fme/ace/train/__main__.py", line 11, in <module>
47     main(yaml_config=args.yaml_config)
48   File "/full-model/fme/fme/ace/train/train.py", line 545, in main
49     run_train_from_config(train_config)
50   File "/full-model/fme/fme/ace/train/train.py", line 529, in run_train_from_config
51     trainer.train()
52   File "/full-model/fme/fme/ace/train/train.py", line 218, in train
53     valid_logs = self.validate_one_epoch()
54   File "/full-model/fme/fme/ace/train/train.py", line 369, in validate_one_epoch
55     ic, normed_ic = self.stepper.get_initial_condition(batch.data)
56   File "/full-model/fme/fme/core/stepper.py", line 519, in get_initial_condition
57     return ic, self.normalizer.normalize(ic)
58   File "/full-model/fme/fme/core/normalizer.py", line 101, in normalize
59     return _normalize(tensors, means=self.means, stds=self.stds)
60 RuntimeError: The following operation failed in the TorchScript interpreter.
61 Traceback of TorchScript (most recent call last):
62   File "/full-model/fme/fme/core/normalizer.py", line 138, in fallback_function
63 ) -> TensorDict:
64     return {
65         k: (t - means[k]) / stds[k] if k in means.keys() else t
66             ~~~~~~~~~~~~ <--- HERE
67         for k, t in tensors.items()
68     }
69 RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
```

**Comment by oliverwm1** [comment]
I'm a bit surprised this is not caught by the[ fine-tuning test](https://github.com/ai2cm/full-model/blob/6517851a23fce52f723a4ee1c860492d015ac809/fme/fme/ace/test_train.py#L361) running on GPUs. Seems there is something different about how `SingleModuleStepperConfig.get_stepper` builds its normalization object versus `ExistingStepperConfig.get_stepper`.

**Comment by oliverwm1** [comment]
Turns out there was a bug in the fine-tuning test, so the "restarted" training actually just resumed from the latest `ckpt.tar` and so when the second training segment started, no epochs were trained since `max_epochs` had already been reached. This is fixed in https://github.com/ai2cm/full-model/pull/1063, but fixing the test is still not reproducing the error I found above even on the GPU CI tests. This is surprising to me.
